torch/distributed/tensor/parallel/style.py

```diff
@@ -405,8 +416,32 @@ def _prepare_input_fn(self, inputs, device_mesh):
                 prepared_inputs.append(inp)
         return tuple(prepared_inputs)
 
+    def _prepare_input_kwarg_fn(self, inputs, kwarg_inputs, device_mesh):
+        prepared_arg_inputs = self._prepare_input_fn(inputs, device_mesh)
+        prepared_kwarg_inputs = {}
+        for kwarg_key in kwarg_inputs.keys():
+            kwarg_val = kwarg_inputs[kwarg_key]
+            input_layout = None
+            if kwarg_key in self.input_kwarg_layouts:
+                input_layout = self.input_kwarg_layouts[kwarg_key]
+                if input_layout is not None:
+                    kwarg_val = DTensor.from_local(kwarg_val, device_mesh, (input_layout,), run_check=False)
+
+            if kwarg_key in self.desired_input_kwarg_layouts:
+                assert kwarg_key in self.input_kwarg_layouts, f"input_kwarg_layouts should have the key {kwarg_key}!"
+                desired_layout = self.desired_input_kwarg_layouts[kwarg_key]
+                if desired_layout is not None and desired_layout != input_layout:
+                    kwarg_val = kwarg_val.redistribute(placements=(desired_layout,))
+
+            prepared_kwarg_inputs[kwarg_key] = kwarg_val.to_local() if self.use_local_output else kwarg_val
```

Comment:
`kwarg_val` can be anything(`None` or whatever type here) and it fails on HF models

Reply:
should be fixed now